src/src/Controller/AbandonedCartController.php

```diff
@@ -0,0 +1,254 @@
+        if ($current_time < $reminder_time) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Send abandoned cart emails to customers.
+     *
+     * @return void
+     */
+    private function send(AbandonedCart $cart): bool
+    {
+        $response = $this->api->triggerAutomation(
+            $this->configuration->get('SMAILY_CART_AUTORESPONDER'),
+            [$this->generatePayload($cart)],
+        );
+
+        if ($response->getStatusCode() !== 200) {
+            Logger::logErrorWithFormatting(
+                'Failed sending out abandoned cart email for email: %s, cart_id: %s. ' .
+                'Smaily response HTTP response code: %s.',
+                $cart->email,
+                $cart->cartID,
+                $response->getStatusCode()
+            );
+
+            return false;
+        }
+
+        $body = json_decode($response->getBody()->getContents(), true);
+        if (!isset($body['code']) || $body['code'] !== 101) {
+            Logger::logErrorWithFormatting(
+                'Failed sending out abandoned cart email for email: %s, cart_id: %s. ' .
+                'Smaily response code: %s, message: %s.',
+                $cart->email,
+                $cart->cartID,
+                $body['code'],
+                $body['message']
+            );
+
+            return false;
+        }
+
+        $sentCart = new SentCart();
+        $sentCart->id_customer = $cart->customerID;
+        $sentCart->id_cart = $cart->cartID;
+        $sentCart->date_sent = date('Y-m-d H:i:s');
+        $sentCart->save();
+
+        return true;
+    }
+
+    private function generatePayload(AbandonedCart $cart): array
+    {
+        $payload = [
+            'email' => $cart->email,
+        ];
+
+        $syncAdditional = json_decode($this->configuration->get('SMAILY_CART_SYNCRONIZE_ADDITIONAL'), true);
+        if (in_array('first_name', $syncAdditional) && $syncAdditional['first_name'] === true) {
+            $payload['first_name'] = $cart->firstName;
+        }
+
+        if (in_array('last_name', $syncAdditional) && $syncAdditional['last_name'] === true) {
+            $payload['last_name'] = $cart->lastName;
+        }
+
+        // Populate abandoned cart with empty values for legacy api.
+        $fields_available = [
+            'base_price',
+            'description',
+            'name',
+            'price',
+            'quantity',
+            'sku',
+        ];
+        foreach ($fields_available as $field) {
+            for ($i = 1; $i <= 10; ++$i) {
+                $payload['product_' . $field . '_' . $i] = '';
+            }
+        }
+
+        $selected_fields = array_intersect($fields_available, array_keys(array_filter($syncAdditional)));
+
+        // Collect products of abandoned cart.
+        $count = 1;
+        $currency = \Context::getContext()->currency->iso_code;
+        foreach ($cart->products as $product) {
```

Comment:
$payload['over_10_products'] = count($cart->products) > 10;
foreach(array_slice($cart->products, 0, 10) as $i => $product) {
...
$payload['product_base_price_' . $i + 1]  = ....; // btw why 0 are missing now?
}

Not sure about $i, it will work only if there are normalized array [0 => {...}, 1 => {...},2 => {...},3 => {...}, ... ]

Reply:
Got it. The main point of my comment was that the $count variable is redundant. Try to avoid unnecessary variables, then your code will be more readable. Here, for example, you can trim the $cart->products array to the limit you need. And work only with it. This will eliminate additional checks inside the foreach loop `if ($count > 10) {`

Reply:
Refactored this one too https://github.com/sendsmaily/smaily-prestashop-module/pull/57/commits/26cfce7b15788ad2a29b3c20b084d8ed1a029bf3